### PR TITLE
[merkle] Add merkle module types

### PR DESCRIPTION
**Pull request by kerol2r20**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by kerol2r20** [comment]
Excuse me, how add add dependencies on @types?
This package depend on @types/node.

File: types/merkle/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Installation
```
Comment (plantain-00) [inline types/merkle/README.md]: The file will be generated automatically when published to npm, so you don't need to add this file as source code.

File: types/merkle/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (plantain-00) [inline types/merkle/tsconfig.json]: For new packages, `strictNullChecks` should be enabled.

File: types/merkle/tslint.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (plantain-00) [inline types/merkle/tslint.json]: For new packages, this file should only `"extends": "dtslint/dt.json"`

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Review by plantain-00 — APPROVED** [review]

**Pull request merged**